### PR TITLE
fix: fixed Bicep warning caused by the identity type

### PR DESCRIPTION
**Pull request by krbar**
## Description

Fixed BCP321 warning caused by the `identity` variable in multiple modules.

image

## Pipeline Reference

| Pipeline |
| -------- |
| [avm.res.api-management.service](https://github.com/krbar/bicep-registry-modules/actions/workflows/avm.res.api-management.service.yml) |
| [avm.res.app.container-app](https://github.com/krbar/bicep-registry-modules/actions/workflows/avm.res.app.container-app.yml) |
| [avm.res.batch.batch-account](https://github.com/krbar/bicep-registry-modules/actions/workflows/avm.res.batch.batch-account.yml) |
| [avm.res.cache.redis](https://github.com/krbar/bicep-registry-modules/actions/workflows/avm.res.cache.redis.yml) |
| [avm.res.data-factory.factory](https://github.com/krbar/bicep-registry-modules/actions/workflows/avm.res.data-factory.factory.yml) |
| [avm.res.databricks.access-connector](https://github.com/krbar/bicep-registry-modules/actions/workflows/avm.res.databricks.access-connector.yml) |
| [avm.res.db-for-postgre-sql.flexible-server](https://github.com/krbar/bicep-registry-modules/actions/workflows/avm.res.db-for-postgre-sql.flexible-server.yml)<br>Error related to the WAF Reliability, not to this PR. Both static and deployment validations work fine. |
| [avm.res.logic.workflow](https://github.com/krbar/bicep-registry-modules/actions/workflows/avm.res.logic.workflow.yml) |
| [avm.res.net-app.net-app-account](https://github.com/krbar/bicep-registry-modules/actions/workflows/avm.res.net-app.net-app-account.yml) |
| [avm.res.storage.storage-account](https://github.com/krbar/bicep-registry-modules/actions/workflows/avm.res.storage.storage-account.yml) |

## Type of Change

- [ ] Update to CI Environment or utlities (Non-module effecting changes)
- [x] Azure Verified Module updates:
  - [x] Bugfix containing backwards compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [ ] Update to documentation

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] I have run `Set-AVMModule` locally to generate the supporting module files.
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings

### CODE REVIEW

**Review by JPEasier — APPROVED** [review]

File: avm/res/api-management/service/main.bicep [hunk 1]
```diff
@@ -133,7 +133,7 @@ var authorizationServerList = !empty(authorizationServers) ? authorizationServer
 var formattedUserAssignedIdentities = reduce(map((managedIdentities.?userAssignedResourceIds ?? []), (id) => { '${id}': {} }), {}, (cur, next) => union(cur, next)) // Converts the flat array to an object like { '${id1}': {}, '${id2}': {} }
 
 var identity = !empty(managedIdentities) ? {
-  type: (managedIdentities.?systemAssigned ?? false) ? (!empty(managedIdentities.?userAssignedResourceIds ?? {}) ? 'SystemAssigned,UserAssigned' : 'SystemAssigned') : (!empty(managedIdentities.?userAssignedResourceIds ?? {}) ? 'UserAssigned' : null)
+  type: (managedIdentities.?systemAssigned ?? false) ? (!empty(managedIdentities.?userAssignedResourceIds ?? {}) ? 'SystemAssigned,UserAssigned' : 'SystemAssigned') : (!empty(managedIdentities.?userAssignedResourceIds ?? {}) ? 'UserAssigned' : 'None')
```
Comment (JPEasier) [inline avm/res/api-management/service/main.bicep:136]: Sinc the last Bicep CLI update I'am facing a second issue regarding the differenc between 
`'SystemAssigned,UserAssigned'` and `'SystemAssigned, UserAssigned'`

```
Warning BCP036: The property "type" expected a value of type "'None' | 'SystemAssigned' | 'SystemAssigned, UserAssigned' | 'UserAssigned' | null" but the provided value in source declaration "identity" is of type "'None' | 'SystemAssigned' | 'SystemAssigned,UserAssigned' | 'UserAssigned'".
```

**Review by JPEasier — DISMISSED** [review]

**Review by segraef — DISMISSED** [review]
🚀

File: avm/res/api-management/service/main.bicep (hunk 1 above)
Comment (krbar) [reply avm/res/api-management/service/main.bicep:136]: @JPEasier  I've seen it in a few modules, but I think that the issue is not the space between `SystemAssigned` and `UserAssigned` but the `null` at the end of the expected value. I have no idea under which condition the `null` should be provided.

Comment (ChrisSidebotham) [reply avm/res/api-management/service/main.bicep:136]: @JPEasier / @krbar - can this be resolved, the PR is approved but this conversation is blocking merge

Comment (AlexanderSehr) [reply avm/res/api-management/service/main.bicep:136]: Yeah would be great to not replace one warning with another one :D @JPEasier do you have an overview for which modules you've experienced the issue? Then I guess we could at least set those to null? Does that make sense? :D

Comment (AlexanderSehr) [reply avm/res/api-management/service/main.bicep:136]: Resolving your comment @JPEasier as you approved the PR ;)

**Review by AlexanderSehr — DISMISSED** [review]

**Review by AlexanderSehr — APPROVED** [review]

**Pull request merged**